scripts/zones/Norg/npcs/_700.lua

```diff
@@ -3,50 +3,74 @@
 -----------------------------------
-require("scripts/globals/missions");
-require("scripts/globals/settings");
+require("scripts/globals/keyitems")
+require("scripts/globals/missions")
+require("scripts/globals/npc_util")
+require("scripts/globals/rhapsodies")
+require("scripts/globals/settings")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end;
+end
 
 function onTrigger(player,npc)
 
-    local ZilartMission = player:getCurrentMission(ZILART);
-    local currentMission = player:getCurrentMission(BASTOK);
-    local ZilartStatus = player:getCharVar("ZilartStatus");
+    local ZilartMission = player:getCurrentMission(ZILART)
+    local currentMission = player:getCurrentMission(BASTOK)
+    local ZilartStatus = player:getCharVar("ZilartStatus")
+    local RhapsodiesMission = player:getCurrentMission(ROV)
 
     -- Checked here to be fair to new players
     local DMEarrings = 0;
     for i=14739, 14743 do
         if (player:hasItem(i)) then
-            DMEarrings = DMEarrings + 1;
+            DMEarrings = DMEarrings + 1
         end
     end
 
     if (ZilartMission == tpz.mission.id.zilart.WELCOME_TNORG) then
-        player:startEvent(2); -- Zilart Missions 2
+        player:startEvent(2) -- Zilart Missions 2
     elseif (ZilartMission == tpz.mission.id.zilart.ROMAEVE and player:getCharVar("ZilartStatus") <= 1) then
-        player:startEvent(3); -- Zilart Missions 9
+        player:startEvent(3) -- Zilart Missions 9
     elseif (ZilartMission == tpz.mission.id.zilart.THE_HALL_OF_THE_GODS) then
-        player:startEvent(169); -- Zilart Missions 11
+        player:startEvent(169) -- Zilart Missions 11
     elseif (currentMission == tpz.mission.id.bastok.THE_PIRATE_S_COVE and player:getCharVar("MissionStatus") == 1) then
-        player:startEvent(98); -- Bastok Mission 6-2
+        player:startEvent(98) -- Bastok Mission 6-2
     elseif (ZilartMission == tpz.mission.id.zilart.THE_SEALED_SHRINE and ZilartStatus == 0 and DMEarrings <= NUMBER_OF_DM_EARRINGS) then
-        player:startEvent(172);
+        player:startEvent(172)
+    elseif RhapsodiesMission == tpz.mission.id.rov.THE_BEGINNING then
+        player:startEvent(276)
+    elseif RhapsodiesMission == tpz.mission.id.rov.FLAMES_OF_PRAYER then
+        player:startEvent(277)
+    elseif RhapsodiesMission == tpz.mission.id.rov.WHAT_LIES_BEYOND then
+        player:startEvent(278)
+    elseif player:getCharVar("ZeidIICipher") == 1 then
+        if npcUtil.giveItem(player, 10160) then -- Cipher: Zeid II
+            player:completeMission(ROV, tpz.mission.id.rov.VOLTO_OSCURO)
+            player:addMission(ROV, tpz.mission.id.rov.RING_MY_BELL)
+            player:setCharVar("ZeidIICipher", 0)
+        end
+    elseif RhapsodiesMission == tpz.mission.id.rov.VOLTO_OSCURO then
+        player:startEvent(279)
+    elseif
+        RhapsodiesMission == tpz.mission.id.rov.RING_MY_BELL and
+        tpz.rhapsodies.charactersAvailable(player)
+    then
```

Comment:
> This was reported to me, but I haven't had time to look into:
> 
> > On Rov  1-18 mission ring my bell .. when i click on door .. in the server console it says error l[21:59:23][Error] luautils::onTrigger: .\scripts/globals/rhapsodies.lua:323: bad argument 1 to 'pairs' (table expected, got nil)     the door gives no cutscene

`RING_MY_BELL` has no entries `tpz.rhapsodies.requiredCharacters`, so shouldn't be calling `tpz.rhapsodies.charactersAvailable` unless characters need to be added to the table. I think this was discussed at some point.

Reply:
Will be resolved by #911